SQL/database_changelog.txt

```diff
@@ -1,3 +1,15 @@
+8th April 2017, by Shadowlight213
+
+Modified table 'player' adding the column 'exp'
+
+ALTER TABLE player ADD exp mediumtext NOT NULL DEFAULT '';
```

Comment:
Alright the explanation I promised: MySQL is a relational database which in essence means it expects each value to be contained to its own column. Having multiple data points in one column is a non-relational way to store data as it's just one big string with delimited values. This can cause you various problems down the line, for a brief overview see: http://stackoverflow.com/questions/3653462/is-storing-a-delimited-list-in-a-database-column-really-that-bad.
(Don't worry about knowing database normalization unless you're interested and have a lot of free time)
As with everything database there's situations where this can be fine but this isn't one.

To better handle this data we can instead store it on a different table and make a reference to one of that table's rows in our current one. Some links on this: http://stackoverflow.com/questions/756370/storing-multiple-employee-ids-in-one-column-of-data http://stackoverflow.com/questions/18603372/how-to-make-sql-many-to-many-same-type-relationship-table.

We already use some tables like this in our schema for polls: Rather than trying to bundle all the options for a poll with the question, the options are instead kept on their own table and store the value `pollid` which points to the `id` value of the row for their question in the `poll_question` table. Each option is a distinct row in `poll_option` but they all point to the same question. And then `poll_vote` containing the actual user responses has a pointer to both `poll_option` and `poll_question`.

I didn't leave myself enough time for this so i'll write you an example of how to implement this and what JOIN does tomorrow.